go/libraries/doltcore/sqle/dolt_diff_table_function.go

```diff
@@ -0,0 +1,458 @@
+// Copyright 2020-2021 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sqle
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/dolthub/dolt/go/libraries/doltcore/diff"
+	"github.com/dolthub/dolt/go/libraries/doltcore/doltdb"
+	"github.com/dolthub/dolt/go/libraries/doltcore/rowconv"
+	"github.com/dolthub/dolt/go/libraries/doltcore/schema"
+	"github.com/dolthub/dolt/go/libraries/doltcore/sqle/dtables"
+	"github.com/dolthub/dolt/go/libraries/doltcore/sqle/sqlutil"
+	"github.com/dolthub/dolt/go/store/types"
+
+	"github.com/dolthub/go-mysql-server/sql"
+)
+
+var _ sql.TableFunction = (*DiffTableFunction)(nil)
+
+type DiffTableFunction struct {
+	ctx            *sql.Context
+	tableNameExpr  sql.Expression
+	toCommitExpr   sql.Expression
+	fromCommitExpr sql.Expression
+	tableName      string
+	toCommitVal    interface{}
```

Comment:
We need these in a couple of different places, but i was able to shift the code around and clean it up so that we don't have to carry the values around in the DTF struct. Looks a little cleaner now.